packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -982,6 +1018,27 @@ export interface IBuildImage {
    */
   readonly defaultComputeType: ComputeType;
 
+  /**
+   * The type of credentials AWS CodeBuild uses to pull images in your build.
+   *
```

Comment:
Wouldn’t it make more sense to use the service role as the default? Feels like that is a more secure default

Reply:
This is not “credentials type”, it’s “principal type”. It basically determines which principal would authenticate against the repository

Reply:
Changed.

Reply:
> Wouldn’t it make more sense to use the service role as the default? Feels like that is a more secure default

Changed.